### PR TITLE
Changed Assign repo->package even when manage_repo=false on Debian+Re…

### PR DESCRIPTION
**Pull request by danielsreichenbach**
#### Pull Request (PR) description
This changes the repository installation to retain the relation between package repository and package even when the repository is not managed by the module.

#### This Pull Request (PR) fixes the following issues

This fixes use issues caused by missing dependency when e.g. using other modules to handle InfluxData packages like InfluxDB, Chronograf or Kapacitor and Telegraf is not managing the repository itself.

### CODE REVIEW

File: manifests/service.pp [hunk 1]
```diff
@@ -4,8 +4,6 @@
 #
 class telegraf::service {
 
-  assert_private()
```
Comment (bastelfreak) [inline manifests/service.pp]: can you explain why you removed this?

Comment (danielsreichenbach) [reply manifests/service.pp]: Sorry, that was actually a facepalm moment. Just removed that.

**Comment by Woth-Ursula** [comment]
> This is Vox Pupuli Tasks, your friendly Vox Pupuli Github Bot. I noticed that your pull request has CI failures. Can you please have a look at the failing CI jobs?

The CI failures contain errors/warning like below. They were already in the module and do not belong to our changes. 
Is it ok when I fix such things? 

`manifests/install.pp:76:manifest_whitespace_closing_brace_before:ERROR:there should be a single space or newline before a closing brace
manifests/install.pp:62:manifest_whitespace_two_empty_lines:ERROR:there should be no two consecutive empty lines
manifests/install.pp:39:strict_indent:WARNING:indent should be 10 chars and is 12`

**Comment by bastelfreak** [comment]
hi, sorry for the trouble! I fixed those in https://github.com/voxpupuli/puppet-telegraf/pull/151. can you please rebase?

**Comment by bastelfreak** [comment]
thanks for the updates. Can you add unit tests to ensure the ordering is now correct?

**Comment by Woth-Ursula** [comment]
What does the "bug" label stands for? Did you find s.th. not working?

**Comment by Woth-Ursula** [comment]
Added some specs. Hope they meet your expectations.

**Comment by bastelfreak** [comment]
thanks for the work!

**Pull request merged**